qutebrowser/browser/webengine/webenginetab.py

```diff
@@ -1225,6 +1233,10 @@ def shutdown(self):
         self._tab.action.exit_fullscreen()
         self._widget.shutdown()
 
+    def run_js_sync(self, code, *, world=None):
+        # pylint: disable=unused-argument
+        return None
```

Comment:
This should `raise browsertab.UnsupportedOperationError` instead, like `clear_ssl_errors` above (and probably the docstring should be adjusted accordingly).

For `networkaccessmanager`, the code using it actually needs `None` as a value to say "there is no NAM available", but for `run_js_sync`, `None` could be a valid return value (and not all callers check the return value), so that could lead to subtle bugs when someone isn't aware that it's QtWebKit-specific.